tests/adapters_t/batchsystems_t/test_slurm.py

```diff
@@ -0,0 +1,205 @@
+
+        self.slurm_adapter = SlurmAdapter()
+
+    def tearDown(self):
+        self.mock_async_run_command.reset_mock()
+
+    def setup_config_mock(self, options=None):
+        self.config = self.mock_config.return_value
+        self.config.BatchSystem.max_age = 10
+        if options:
+            self.config.BatchSystem.options = options
+        else:
+            self.config.BatchSystem.options = {}
+
+    def test_disintegrate_machine(self):
+        self.assertIsNone(
+            run_async(self.slurm_adapter.disintegrate_machine, drone_uuid="test")
+        )
+
+    def test_drain_machine(self):
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+
+        self.mock_async_run_command.reset_mock()
+
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+        self.assertIsNone(
+            run_async(self.slurm_adapter.drain_machine, drone_uuid="not_exists")
+        )
+        self.mock_async_run_command.side_effect = CommandExecutionFailure(
+            message="Does not exists", exit_code=1, stderr="Does not exists"
+        )
+        with self.assertRaises(CommandExecutionFailure):
+            self.assertIsNone(
+                run_async(self.slurm_adapter.drain_machine, drone_uuid="idle_machine")
+            )
+
+        self.mock_async_run_command.side_effect = None
+
+    def test_drain_machine_without_options(self):
+        self.setup_config_mock()
+        self.slurm_adapter = SlurmAdapter()
+
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+
+    def test_integrate_machine(self):
+        self.assertIsNone(
+            run_async(self.slurm_adapter.integrate_machine, drone_uuid="VM-1")
+        )
+
+    def test_get_resource_ratios(self):
+        self.assertEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
+        self.mock_async_run_command.assert_called_with(self.command)
+
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="not_exists"),
+            {},
+        )
+
+    def test_get_resource_ratios_without_options(self):
+        self.setup_config_mock()
+        del self.config.BatchSystem.options
+        self.slurm_adapter = SlurmAdapter()
+
+        self.assertCountEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
```

Comment:
Is there a reason for using assertCountEqual here and asserEqual before in line 123 for an almost identical test?